test/e2e/far_e2e_test.go

```diff
@@ -0,0 +1,208 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	apiErrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/medik8s/fence-agents-remediation/api/v1alpha1"
+	farController "github.com/medik8s/fence-agents-remediation/controllers"
+	"github.com/medik8s/fence-agents-remediation/pkg/cli"
+	farUtils "github.com/medik8s/fence-agents-remediation/test/e2e/utils"
+)
+
+const (
+	// testNamespace  = "fence-agents-remediation"
+	testNamespace  = "openshift-operators"
+	fenceAgentIPMI = "fence_ipmilan"
+
+	// eventually parameters
+	timeout      = 2 * time.Minute
+	pollInterval = 10 * time.Second
+	offsetExpect = 1
+)
+
+var _ = Describe("FAR E2e", func() {
+	testShareParam := map[v1alpha1.ParameterName]string{
+		"--username": "admin",
+		"--password": "password",
+		"--action":   "reboot",
+		"--ip":       "192.168.111.1",
+		"--lanplus":  "",
+	}
+	testNodeParam := map[v1alpha1.ParameterName]map[v1alpha1.NodeName]string{
+		"--ipport": {
+			"master-0": "6230",
+			"master-1": "6231",
+			"master-2": "6232",
+			"worker-0": "6233",
+			"worker-1": "6234",
+			"worker-2": "6235",
+		},
+	} // get ports
+	Context("fence agent - fence_ipmilan", func() {
+		var (
+			nodeBootTimeBefore metav1.Time
+			far                *v1alpha1.FenceAgentsRemediation
+			cond               corev1.NodeCondition
+			errBoot            error
+		)
+		testNode := &corev1.Node{}
+		nodes := &corev1.NodeList{}
+		BeforeEach(func() {
+			Expect(k8sClient.List(context.Background(), nodes, &client.ListOptions{})).ToNot(HaveOccurred())
+			if len(nodes.Items) <= 1 {
+				Fail("there is one or less available nodes in the cluster")
+			}
+			//TODO: Randomize the node selection
+			// Use FA on the first node - master-0
+			testNode = &nodes.Items[0]
+			log.Info("Testing Node", "Node name", testNode.Name)
+
+			// save the node's boot time prior to the fence agent call
+			if cond, errBoot = getKubeletReadyCondition(testNode.Name); errBoot != nil {
+				log.Error(errBoot, "Can't get boot time of the node")
+			}
+			nodeBootTimeBefore = cond.LastTransitionTime
+			far = createFAR(testNode.Name, fenceAgentIPMI, testShareParam, testNodeParam)
+		})
+
+		AfterEach(func() {
+			deleteFAR(far)
+		})
+
+		When("running FAR to reboot node ", func() {
+			It("should execute the fence agent cli command", func() {
+				By("checking the CR has been created")
+				farCR := &v1alpha1.FenceAgentsRemediation{}
+				ExpectWithOffset(offsetExpect, k8sClient.Get(context.Background(),
+					client.ObjectKey{Name: testNode.Name, Namespace: testNamespace}, farCR)).ToNot(HaveOccurred())
+
+				By("checking the command has been executed successfully")
+				checkFarLogs(cli.SuccessCommandLog)
+
+				By("checking the node's boot time after running the FA")
+				if !nodeBootTimeBefore.IsZero() {
```

Comment:
I wonder if it would be better to check this value at the beginning of the section, so to avoid all the previous steps

Reply:
I agree with that, but IIUC the value of `nodeBootTimeBefore` is set before anything else in `BeforeEach`, and if zero all the test has done is useless.

Reply:
On a second though, I will just test this value in the _BeforeEach_ and fail if it is zero.